### PR TITLE
core/dbus-service: refuse bind mounting over /run/credentials/

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: src/core/dbus-service.c [hunk 1]
```diff
@@ -174,26 +177,25 @@ static int bus_service_method_mount(sd_bus_message *message, void *userdata, sd_
...
         /* Ensure that the unit was started in a private mount namespace */
         if (!exec_needs_mount_namespace(c, NULL, unit_get_exec_runtime(u)))
                 return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Unit not running in private mount namespace, cannot activate bind mount");
 
-        PidRef* unit_pid = unit_main_pid(u);
-        if (!pidref_is_set(unit_pid) || !UNIT_IS_ACTIVE_OR_RELOADING(unit_active_state(u)))
-                return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Unit is not running");
+        /* If it would be dropped at startup time, return an error. */
+        if (path_startswith_strv(dest, c->inaccessible_paths))
+                return sd_bus_error_setf(error, SD_BUS_ERROR_FILE_EXISTS, "%s is not accessible to this unit", dest);
```
Comment (yuwata) [inline src/core/dbus-service.c]: The error code and message sound conflict each other. Maybe, SD_BUS_ERROR_ACCESS_DENIED is better?

File: src/core/exec-credential.c [hunk 2]
```diff
@@ -242,6 +242,19 @@ bool exec_context_has_encrypted_credentials(const ExecContext *c) {
...
+        const char *e;
+
+        assert(runtime_prefix);
+        assert(path);
+
+        e = path_startswith(path, runtime_prefix);
+        if (!e)
+                return false;
+
+        return !isempty(path_startswith(e, "credentials"));
```
Comment (yuwata) [inline src/core/exec-credential.c]: Hm? /run/credentials itself is allowed? But /run/credentials/foo is refused??

File: src/core/dbus-service.c [hunk 3]
```diff
@@ -6,6 +6,7 @@
 #include "async.h"
 #include "bus-common-errors.h"
 #include "bus-get-properties.h"
+#include "exec-credential.h"
```
Comment (yuwata) [inline src/core/dbus-service.c]: Not sorted?

File: src/core/mount.c [hunk 4]
```diff
@@ -8,6 +8,7 @@
 #include "sd-messages.h"
 
 #include "alloc-util.h"
+#include "exec-credential.h"
```
Comment (yuwata) [inline src/core/mount.c]: Also here.

File: src/core/mount.c [hunk 5]
```diff
@@ -425,22 +426,6 @@ static bool mount_is_extrinsic(Unit *u) {
...
 }
 
-static bool mount_point_is_credentials(Manager *manager, const char *path) {
-        const char *e;
-
-        assert(manager);
-        assert(path);
-
-        /* Returns true if this is a credentials mount. We don't want to generate mount units for them,
-         * since their lifetime is strictly bound to services. */
```
Comment (yuwata) [inline src/core/mount.c:435]: Please keep the comment.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/core/mount.c (hunk 5 above)
Comment (YHNdnzj) [reply src/core/mount.c:435]: This is now merged into `mount_setup_unit`. I don't think we should carry this in `exec-credential.c`...

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**